ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/deduplication/DeduplicationPeriodService.scala

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.deduplication
+
+import java.time.{Duration, Instant}
+
+import akka.stream.Materializer
+import com.daml.error.ContextualizedErrorLogger
+import com.daml.ledger.api.DeduplicationPeriod
+import com.daml.ledger.api.domain.ApplicationId
+import com.daml.lf.data.Ref
+import com.daml.logging.LoggingContext
+import com.daml.platform.server.api.validation.{DeduplicationPeriodValidator, ErrorFactories}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeduplicationPeriodService(
```

Comment:
Can this be inlined into the `WriteServiceWithDeduplicationOffsetSupport`?

Reply:
I agree, or I'd rather avoid calling it `Service` as it clashes with the participant naming (also because we don't really have strict layering in KV).

Reply:
I changed the naming. I tried inlining it in fact but the WriteServiceWithDeduplicationSupport increased in size quite a bit and it muddies the responsibility of the WriteService itself.